app/controllers/api/checkout_controller.rb

```diff
@@ -133,4 +124,12 @@ def authenticate_checkout
       nil
     end
   end
+
+  def authenticate_card
+    @uuid = params[:uuid]
+    @card = CheckoutCard.find_by(uuid: @uuid)
```

Comment:
Should first check if card exists otherwise return 404. @card.active gives an error now since @card is nil in certain scenarios